marimbabot_planning/include/marimbabot_planning/planning.h

```diff
@@ -31,6 +31,10 @@ class Planning
         // MoveIt! MoveGroupInterface
         moveit::planning_interface::MoveGroupInterface move_group_interface_;
 
+        // Describes when a "hit" was performed in terms of 
+        // how long after the start of the trajectory this point should be reached 
+        std::list<ros::Duration> hit_plan_durations;
```

Comment:
Doing this as a side effect of a program that is otherwise mostly [pure](https://en.wikipedia.org/wiki/Pure_function) can lead to issues, as nobody expects that the trajectory needs to be generated in order. One could e.g. generate the approach to the initial note in the end.